openerp/openupgrade/openupgrade.py

```diff
@@ -622,3 +625,167 @@ def wrapped_function(cr, version):
                 raise
         return wrapped_function
     return wrap
+
+
+def remove_sql_constraint_duplicates(cr, model, constraint_attrs):
+    """
+    Remove all duplicates after a sql constraint is applied on a model.
+
+    For every field many2one and many2many with the given model as relation,
+    change the duplicate ids with the id of the record kept.
+
+    This script must be called in post-migration so that the model being
+    edited can be accessed through the orm.
+
+    When upgrading the module, if there are duplicates, integrity errors
+    will be raised before the script is run but this will not prevent
+    the script from running.
+
```

Comment:
Can you warn about the information loss that might occur? As in 'Use with care, ...'